pkg/controller/jobs/pod/pod_controller_test.go

```diff
@@ -2920,6 +2919,178 @@ func TestReconciler(t *testing.T) {
 				},
 			},
 		},
+		"the failed pods are finalized in order": {
```

Comment:
Add a unit test for not finalizing pods when waiting for expectations.

Another one where there are 2 Pods, 1 Pod already has a finalizer removed and there is 1 replacement Pod. No new pod should get a finalizer removed.